### PR TITLE
Demos for date and time pickers now modern-friendly

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**
- Added input box.
- Added `skyId` directive.
- Stackblitz demos should now load modern version when `svcid=skyux-modern` URL param is present.

### CODE REVIEW

File: src/app/public/modules/datepicker/datepicker.component.spec.ts [hunk 1]
```diff
@@ -1385,13 +1385,11 @@ describe('datepicker', () => {
...
-
-      const inputGroupEl = inputBoxEl.querySelector('.sky-form-group > .sky-input-group');
-      const inputEl = inputGroupEl.children.item(0);
-      const inputGroupBtnEl1 = inputGroupEl.children.item(1);
+      const inputEl = inputBoxEl.querySelector('input');
+      const inputGroupBtnEl = inputBoxEl.querySelector('.sky-input-group-btn > button');
 
       expect(inputEl).toHaveCssClass('input-box-datepicker-input');
-      expect(inputGroupBtnEl1.children.item(0)).toHaveCssClass('sky-input-group-datepicker-btn');
+      expect(inputGroupBtnEl).toHaveCssClass('sky-input-group-datepicker-btn');
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/datepicker/datepicker.component.spec.ts:1392]: ☝️ Not part of this PR. However, this fixes changes to the input-box introduced in `"@skyux/forms": "4.13.0"`

File: src/app/public/modules/timepicker/timepicker.component.spec.ts [hunk 2]
```diff
@@ -820,13 +820,11 @@ describe('Timepicker', () => {
...
-
-      const inputGroupEl = inputBoxEl.querySelector('.sky-form-group > .sky-input-group');
-      const inputEl = inputGroupEl.children.item(0);
-      const inputGroupBtnEl1 = inputGroupEl.children.item(1);
+      const inputEl = inputBoxEl.querySelector('input');
+      const inputGroupBtnEl = inputBoxEl.querySelector('.sky-input-group-btn > button');
 
       expect(inputEl).toHaveCssClass('input-box-timepicker-input');
-      expect(inputGroupBtnEl1.children.item(0)).toHaveCssClass('sky-input-group-timepicker-btn');
+      expect(inputGroupBtnEl).toHaveCssClass('sky-input-group-timepicker-btn');
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/timepicker/timepicker.component.spec.ts:827]: ☝️ Not part of this PR. However, this fixes changes to the input-box introduced in `"@skyux/forms": "4.13.0"`

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-modern-pickers](https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-modern-pickers)

**Review by Blackbaud-PaulCrowder — APPROVED** [review]
